ValveResourceFormat/IO/MaterialExtract.cs

```diff
@@ -0,0 +1,482 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Numerics;
+using System.Text;
+using ValveKeyValue;
+using ValveResourceFormat.Blocks;
+using ValveResourceFormat.ResourceTypes;
+
+namespace ValveResourceFormat.IO;
+
+using ChannelMapping = ValueTuple<Channel, string>;
+
+public struct UnpackInfo
+{
+    public string TextureType { get; init; }
+    public string FileName { get; init; }
+    public Channel Channel { get; init; }
+}
+
+public enum Channel
+{
+    R,
+    G,
+    B,
+    A,
+    _OneChannel,
+    GA,
+    _TwoChannels,
+    RGB,
+    All
```

Comment:
Call it RGBA?

It also feels like it should be possible to make this enum flags, and get rid of _OneChannel/_TwoChannels